bin/check-smart-tests.rb

```diff
@@ -0,0 +1,179 @@
+         default: '/usr/sbin/smartctl',
+         description: 'Path to smartctl executable'
+  option :devices,
+         long: '--devices *DEVICES',
+         short: '-d *DEVICES',
+         default: 'all',
+         description: 'Comma-separated list of devices to check, i.e. "/dev/sda,/dev/sdb"'
+  option :short_test_interval,
+         long: '--short_test_interval INTERVAL',
+         short: '-s INTERVAL',
+         description: 'If more time then this value passed since last short test run, then warning will be raised'
+  option :long_test_interval,
+         long: '--long_test_interval INTERVAL',
+         short: '-l INTERVAL',
+         description: 'If more time then this value passed since last extedned test run, then warning will be raised'
+
+  def initialize
+    super
+    @devices = []
+    @warnings = []
+    @criticals = []
+    set_devices
+  end
+
+  def set_devices
+    if config[:devices] == 'all'
+      `lsblk -plnd -o NAME`.split.each do |name|
+        unless name =~ /\/dev\/loop.*/
+          dev = Device.new(name, config[:executable])
+          @devices.push(dev)
+        end
+      end
+    else
+      config[:devices].split(',').each do |name|
+        dev = Device.new(name, config[:executable])
+        @devices.push(dev)
+      end
+    end
+  end
+
+  def check_tests(dev)
+    if dev.str.empty?
+      @warnings << "#{dev.name}: No self-tests have been logged."
+      return
+    end
+
+    unless dev.str[0]['status'] == 'Completed without error' || dev.str[0]['status'] =~ /Self-test routine in progress/
+      @criticals << "#{dev.name}: Last test failed - #{dev.str[0]['status']}"
+    end
+
+    unless config[:short_test_interval].nil?
+      dev.str.each_with_index do |t, i|
+        if t['test_description'] != 'Short offline'
+          if i == dev.str.length - 1
+            @warnings << "#{dev.name}: No short tests were run for this device in last #{dev.str.length} executions"
+          end
+          next
+        else
+          if dev.pwh.to_i - t['lifetime'].to_i > config[:short_test_interval].to_i
+            @warnings << "#{dev.name}: More than #{config[:short_test_interval]} hours passed since the last short test"
+          end
+          break
+        end
+      end
+    end
+
+    unless config[:long_test_interval].nil?
+      dev.str.each_with_index do |t, i|
+        if t['test_description'] != 'Extended offline'
+          if i == dev.str.length - 1
+            @warnings << "#{dev.name}: No extended tests were run for this device in last #{dev.str.length} executions"
+          end
+          next
+        else
+          if dev.pwh.to_i - t['lifetime'].to_i > config[:long_test_interval].to_i
+            @warnings << "#{dev.name}: More than #{config[:long_test_interval]} hours passed since the last extended test"
+          end
+          break
+        end
+      end
+    end
+  end
+
+  def run
+    @devices.each do |device|
+      check_tests(device)
+    end
+
+    if !@criticals.empty?
```

Comment:
Rather than checking if its not empty you can ask if there is anything in it like this `@criticals.any?`